app/v2/services/PersistenceService.scala

```diff
@@ -226,12 +229,34 @@ class PersistenceServiceImpl @Inject() (persistenceConnector: PersistenceConnect
   ): EitherT[Future, PersistenceError, PaginationMovementSummary] = EitherT {
     persistenceConnector
       .getMovements(eori, movementType, updatedSince, movementEORI, movementReferenceNumber, page, count, receivedUntil, localReferenceNumber)
-      .map(Right(_))
+      .map {
+        summary => movements(page, summary)
+      }
       .recover {
         case NonFatal(thr) => Left(PersistenceError.UnexpectedError(Some(thr)))
       }
   }
 
+  private def isEmptyPage(page: Option[PageNumber], numberOfItemsForPage: Int): Boolean =
+    (page, numberOfItemsForPage) match {
+      case (Some(PageNumber(1)), _)                                     => false
+      case (Some(_), numberOfItemsForPage) if numberOfItemsForPage == 0 => true
+      case (None, _)                                                    => true
```

Comment:
Isn't `None` the same as page 1 -- in which case, this should also be false?

Reply:
Not sure, but if that's the case will change.